### PR TITLE
Add js OOP Frogger homework

### PR DESCRIPTION
**Pull request by Roman-Halenko**
### JS OOP (Frogger)

[Demo](https://roman-halenko.github.io/js-oop-frogger/) |
[Code base](https://github.com/Roman-Halenko/js-oop-frogger)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Roman-Halenko/js-oop-frogger/js/app.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    this.x = 202;
+    this.y = 400;
+};
+
+Player.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.toBegin = function(d) {
+    if (d === 0) {
```
Comment (AMashoshyna) [inline submissions/Roman-Halenko/js-oop-frogger/js/app.js]: Could you please explain, what this method is doing? What is `d`?

File: submissions/Roman-Halenko/js-oop-frogger/js/app.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }, d);
+    }
+};
+
+Player.prototype.update = function() {
+    if (this.y === -20) {
+        this.toBegin(400);
+    }
+    allEnemies.forEach( (e) => {
+        if (e.y === player.y && (player.x <= e.x+80 && player.x+90 >= e.x+15)) {
```
Comment (AMashoshyna) [inline submissions/Roman-Halenko/js-oop-frogger/js/app.js]: For the sake of readability I would put this condition into a named constant (something like "isCollision" or else), but this is just a suggestion.

File: submissions/Roman-Halenko/js-oop-frogger/js/app.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
+// Enemies our player must avoid
+var Enemy = function(x, y) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.x = x;
+    this.y = y;
+    this.speed = randomSpeed(80, 290);
```
[the remark this answers is not in the document]
Comment (Roman-Halenko) [reply submissions/Roman-Halenko/js-oop-frogger/js/app.js]: Using random speed function?

[the remark this answers is not in the document]
Comment (AMashoshyna) [reply submissions/Roman-Halenko/js-oop-frogger/js/app.js]: Putting this logic into a reusable function

File: submissions/Roman-Halenko/js-oop-frogger/js/app.js [hunk 4]
```diff
@@ -58,46 +62,51 @@ Player.prototype.toBegin = function(d) {
...
-    allEnemies.forEach( (e) => {
-        if (e.y === player.y && (player.x <= e.x+80 && player.x+90 >= e.x+15)) {
-            this.toBegin(0);
+    this.collision();
+};
+
+Player.prototype.collision = function() {
+    allEnemies.forEach( e => {
+        let that = this;
+        if (e.y === that.y && that.x <= e.x+80 && that.x+90 >= e.x+15) {
```
Comment (AMashoshyna) [inline submissions/Roman-Halenko/js-oop-frogger/js/app.js]: What are 80, 90, 15? Are those player measurements?  If so, maybe you could put them into a variable similar to "edge"

File: submissions/Roman-Halenko/js-oop-frogger/js/app.js [hunk 5]
```diff
@@ -58,46 +62,51 @@ Player.prototype.toBegin = function(d) {
...
+    if (this.y === this.edge.top) {
         this.toBegin(400);
     }
-    allEnemies.forEach( (e) => {
-        if (e.y === player.y && (player.x <= e.x+80 && player.x+90 >= e.x+15)) {
-            this.toBegin(0);
+    this.collision();
+};
+
+Player.prototype.collision = function() {
```
Comment (AMashoshyna) [inline submissions/Roman-Halenko/js-oop-frogger/js/app.js]: "checkForCollision" maybe? "collision" does not describe what the function is doing. Besides, functions are usually named with verbs.

**Comment by AMashoshyna** [comment]
Nice job! Please see few more comments.

**Comment by Roman-Halenko** [comment]
@AMashoshyna, big thank for your attention :) is my implementation satisfactory now?

**Comment by AMashoshyna** [comment]
Perfect! Ready to be merged.

**Pull request merged**